src/node.cc

```diff
@@ -2364,7 +2366,8 @@ static void DLOpen(const FunctionCallbackInfo<Value>& args) {
   modlist_addon = mp;
 
   Local<String> exports_string = env->exports_string();
-  Local<Object> exports = module->Get(exports_string)->ToObject(env->isolate());
+  Local<Object> exports =
+      module->Get(env->context(), exports_string).ToLocalChecked().As<Object>();
```

Comment:
I think here we don't know if `module.exports` is an object yet? The `Local<Value>` returned by `.ToLocalChecked()` should go through `->ToObject(context).ToLocalChecked()` again.

Reply:
@joyeecheung Thank you for review.

You mean this changes

```
  Local<Object> exports =
      module->Get(env->context(), exports_string).ToLocalChecked()
          ->ToObject(env->context()).ToLocalChecked();
```

or this ?

```
  Local<Object> exports =
      module->Get(exports_string)->ToObject(env->context()).ToLocalChecked();
```

I think A is better because `Local<Value> Get(Local<Value> key)` will be deprecated.

Reply:
@Leko (A) seems better, but I think this is a situation similar to the `vm` one: You can actually get this to crash with the current code, and `.ToLocalChecked()` is not going to change that:

```js
// The path here is just an example, any addon will work
process.dlopen({ exports: undefined }, './test/addons/hello-world/build/Release/binding.node');
```

because calling `ToObject()` on `undefined` gives an exception and therefore an empty `exports` variable.

I think we want to do a similar thing to what we're doing in the vm module: If `Get()` or `ToObject()` return an empty `MaybeLocal`, we should call `dlib.Close()` to clean up the library and return from the function as soon as possible.

Reply:
Thank you @addaleax .
I fixed it by d7cc341 .

Is my understanding correct ?

Reply:
@l2dy yes, this seems about right! Thank you!

Reply:
Not at all !
Thanks for your quick reply.